elastix_napari/elastix_registration.py

```diff
@@ -127,8 +129,14 @@ def elastix_registration(fixed: 'napari.types.ImageData',
 
     # Casting to numpy is currently necessary
     # because of napari's type ambiguity.
-    fixed = np.asarray(fixed).astype(np.float32)
-    moving = np.asarray(moving).astype(np.float32)
+    print(type(fixed))
+    print(fixed.metadata)
+    print(fixed.translate)
+    fixed = image_from_image_layer(fixed)
+    moving = image_from_image_layer(moving)
+    print(type(fixed))
```

Comment:
@ViktorvdValk here we can call:

```
fixed = fixed.astype(itk.F)
```

to cast to an itk.Image[itk.F, D]. Note that `fixed = fixed.astype(np.float32)` does the same.

Reply:
Ah, thanks, that was quite easy :)